demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -50,530 +136,130 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
     return true;
 }
 
-void frameToBlob(const cv::Mat& frame,
-                 InferRequest::Ptr& inferRequest,
-                 const std::string& inputName) {
-    if (FLAGS_auto_resize) {
-        /* Just set input blob containing read image. Resize and layout conversion will be done automatically */
-        inferRequest->SetBlob(inputName, wrapMat2Blob(frame));
-    } else {
-        /* Resize and copy data from the image to the input blob */
-        Blob::Ptr frameBlob = inferRequest->GetBlob(inputName);
-        matU8ToBlob<uint8_t>(frame, frameBlob);
-    }
-}
+void paintInfo(cv::Mat& frame, const PipelineBase::PerformanceInfo& info) {
+    std::ostringstream out;
 
-enum class ExecutionMode {USER_SPECIFIED, MIN_LATENCY};
+    out.str("");
+    out << "FPS:" << std::fixed << std::setprecision(0) << std::setw(3) << info.movingAverageFPS
+        << " (" << std::setprecision(1) << info.FPS << ")";
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 22), cv::FONT_HERSHEY_TRIPLEX, 0.6, cv::Scalar(10, 200, 10), 2);
 
-ExecutionMode getOtherMode(ExecutionMode mode) {
-    return mode == ExecutionMode::USER_SPECIFIED ? ExecutionMode::MIN_LATENCY : ExecutionMode::USER_SPECIFIED;
-}
+    out.str("");
+    out << "Avg Latency:" << std::fixed << std::setprecision(0) << std::setw(4) << info.movingAverageLatencyMs
+        << " (" << std::setprecision(1) << info.getTotalAverageLatencyMs() << ") ms";
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 44), cv::FONT_HERSHEY_TRIPLEX, 0.6, cv::Scalar(200, 10, 10), 2);
 
-void switchMode(ExecutionMode& mode) {
-    mode = getOtherMode(mode);
+    out.str("");
+    out << "Inference Latency:" << std::fixed << std::setprecision(0) << std::setw(4) << info.getLastInferenceLatencyMs() << " ms";
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 66), cv::FONT_HERSHEY_TRIPLEX, 0.6, cv::Scalar(200, 10, 10), 2);
+
+    out.str("");
+    out << "Pool: " << std::fixed << std::setprecision(1) <<
+        info.numRequestsInUse << "/" << FLAGS_nireq;
```

Comment:
This metric cheats. It always reports at least one inferRequest busy even when you have 1 nireq. But you can't keep it always busy. The proper solution would be to run a dedicated thread and sample `RequestsPool`'s `numRequestsInUse`. So I still want to remove the metric.

Reply:
in this particular moment 1 request is busy as we just submitted the data. The metric is useful to see if we have too small/too large requests pool.

Reply:
not relevant any more